src/api.mjs

```diff
@@ -68,16 +68,19 @@ export async function request(url, method, body, headers, signal) {
     );
   }
 
-  let data;
-  try {
-    data = JSON.parse(answer);
-  } catch (err) {
-    throw new Error(
-      `Encountered error when trying to parse JSON body result: "${answer}", error: "${err.toString()}"`
-    );
+  if (results.headers.get("Content-Type")?.toLowerCase().includes("json")) {
```

Comment:
Mhh, I find this statement rather difficult to reason about as I'm not sure if it is a strict convention of setting "Content-type" to include "json" when JSON is passed in the body. E.g. I think there can be servers that omit the Content-Type but still send valid JSON. It'd not good practice but I think ppl anyways do it.

Still, your PR addresses an important issue which is that if we cannot parse the HTTP response body, then we throw an error instead of potentially reporting `answer`.

I think if we're going this route, we may need to start considering doing actual [content negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation), e.g. we reject a valid JSON body if `Content-Type !== application/json` if we've explicitly asked before using the `Accept: application/json` header.

But for now, this PR addresses the issue and so we should merge it.

Reply:
opened issue: https://github.com/neume-network/extraction-worker/issues/51

Reply:
I now understand why I dislike the change. Independent of Content-Type we should try to parse the body as a json and only if it then fails but the content type isn't application/json we should consider forwarding the raw text

Reply:
here are test cases I'd like to see:

- content type: application json and body is valid parsable json then return parsed json
- content type: application json and body is unparsable json then throw error
- content type undefined or could be json as e.g. plain/text and body is valid parsable json then return parsed json
- content type is undefined or could be json but body isn't parsable json then if e.g. plain/text return text otherwise throw not implemented error
- content type is application/html then throw unimplemented error